nautobot/core/views/mixins.py

```diff
@@ -942,15 +944,32 @@ def _get_bulk_edit_delete_all_queryset(self, request):
             queryset = model.objects.all()
         return queryset
 
-    def _bulk_delete_all_context(self, request, queryset):
-        model = queryset.model
-        return {
-            "obj_type_plural": model._meta.verbose_name_plural,
-            "return_url": self.get_return_url(request),
-            "total_objs_to_delete": queryset.count(),
-            "delete_all": True,
-            "table": None,
-        }
+    def send_bulk_delete_objects_to_job(self, request, pk_list, model, delete_all):
+        """Prepare and enqueue bulk delete job."""
+        job_model = Job.objects.get_for_class_path(BulkDeleteObjects.class_path)
+
+        if filterset_class := lookup.get_filterset_for_model(model):
+            filter_query_params = normalize_querydict(request.GET, filterset=filterset_class())
+        else:
+            filter_query_params = None
```

Comment:
Should we log and/or raise an error if there are query params in `request.GET` but we can't find the filterset?

Reply:
There is no direct way of archiving this, as request.GET could include a lot of things, e.g per_page, max_page, tab etc. and may not be a queryset filter params